src/EditorFeatures/Core/InlineRename/CommandHandlers/AbstractRenameCommandHandler_ReturnHandler.cs

```diff
@@ -27,7 +28,7 @@ public bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext co
 
     protected virtual void CommitAndSetFocus(InlineRenameSession activeSession, ITextView textView, IUIThreadOperationContext operationContext)
     {
-        Commit(operationContext);
+        _ = activeSession.CommitAsync(previewChanges: false, operationContext).ReportNonFatalErrorAsync();
```

Comment:
same issues with tracking token.  instead o fthis, we could just have a sync helper that everything calls that does this consistently for all callers.